.github/workflows/build.yml

```diff
@@ -68,25 +71,57 @@ jobs:
         NeedUpdate=0
         if [ "$EXTERNALS_HASH" != "$(cat ./src/Misc/contentHash/externals/${{ matrix.runtime }})" ] ;then
           echo Hash mismatch, Update ./src/Misc/contentHash/externals/${{ matrix.runtime }} to $EXTERNALS_HASH
+
+          echo "EXTERNAL_HASH=$EXTERNALS_HASH" >> $GITHUB_OUTPUT
           NeedUpdate=1
         fi
 
         if [ "$DOTNET_RUNTIME_HASH" != "$(cat ./src/Misc/contentHash/dotnetRuntime/${{ matrix.runtime }})" ] ;then
           echo Hash mismatch, Update ./src/Misc/contentHash/dotnetRuntime/${{ matrix.runtime }} to $DOTNET_RUNTIME_HASH
+
+          echo "DOTNET_RUNTIME_HASH=$DOTNET_RUNTIME_HASH" >> $GITHUB_OUTPUT
           NeedUpdate=1
         fi
 
         exit $NeedUpdate
       env:
         DOTNET_RUNTIME_HASH: ${{hashFiles('**/_layout_trims/runtime/**/*')}}
         EXTERNALS_HASH: ${{hashFiles('**/_layout_trims/externals/**/*')}}
+    - name: update hash
+      if: github.event_name == 'pull_request' && steps.compute-hash.outcome != 'success' 
+      shell: bash
+      run: |
+        ExternalHash=${{ steps.compute-hash.outputs.EXTERNAL_HASH }}
+        DotNetRuntimeHash=${{ steps.compute-hash.outputs.DOTNET_RUNTIME_HASH }}
 
+        if [ -n "$ExternalHash" ]; then
+          echo "$ExternalHash" > ./src/Misc/contentHash/externals/${{ matrix.runtime }}
+        fi
+
+        if [ -n "$DotNetRuntimeHash" ]; then
+          echo "$DotNetRuntimeHash" > ./src/Misc/contentHash/dotnetRuntime/${{ matrix.runtime }}
+        fi
+    - name: cache updated hashes
+      if: github.event_name == 'pull_request' && steps.compute-hash.outcome != 'success' 
+      uses: actions/cache/save@v3
+      with:
+        path: |
+          ./src/Misc/contentHash/externals/${{ matrix.runtime }}
+          ./src/Misc/contentHash/dotnetRuntime/${{ matrix.runtime }}
+        key: compute-hashes-${{ matrix.runtime }}-${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}
+    - name: Create an warning annotation if computed hashes need to be updated        
+      if: github.event_name == 'pull_request' && steps.compute-hash.outcome != 'success' 
+      shell: bash
+      run: |
+        echo "::warning ::Computed hashes do not match, we will automatically update these for you, you can safely ignore the errors on this job"
+        # We need to exit with a failure code so the commit can be process through
+        exit 1
```

Comment:
This is actually required, and _will_ unfortunately cause two independent errors in the run. We need to know that the job has failed, and the failure in `compute-hash` has `continue-on-error` set, which means that if there isn't a second `exit 1` this job will be marked as successful. 
<img width="1407" alt="Screenshot 2023-08-25 at 2 47 14 PM" src="https://github.com/actions/runner/assets/749475/bac6cdd5-3c43-4bc4-9cf4-df8484e8530b">

The way around this, would be to instead of `exit $NEEDS_UPDATE` in `compute-hash` to just output that variable for other steps to pickup, @TingluoHuang what would be your preference on this?

Reply:
I changed this to just use an output variable to reduce the error's for readability